ui/src/app/shared/components/revision.tsx

```diff
@@ -1,9 +1,12 @@
 import * as React from 'react';
 import {revisionUrl} from './urls';
 
-export const Revision = ({repoUrl, revision, children}: {repoUrl: string; revision: string; children?: React.ReactNode}) => {
+export const Revision = ({repoUrl, revision, path, children}: {repoUrl: string; revision: string; path?: string; children?: React.ReactNode}) => {
     revision = revision || '';
-    const url = revisionUrl(repoUrl, revision);
+    let url = revisionUrl(repoUrl, revision);
+    if (path) {
```

Comment:
I think maybe this should be `if (path && path !== ".")`, since an empty path and a `.` path are equivalen.